packages/build/src/plugins_core/deploy/buildbot_client.js

```diff
@@ -0,0 +1,86 @@
+const net = require('net')
+const { promisify } = require('util')
+
+const pEvent = require('p-event')
+
+const { addErrorInfo } = require('../../error/info')
+
+const startBuildbotClient = async function(buildbotServerSocket) {
+  if (buildbotServerSocket === undefined) {
+    return
+  }
+  try {
+    const buildbotClient = net.createConnection(buildbotServerSocket)
+    await pEvent(buildbotClient, 'connect')
+    return buildbotClient
+  } catch (error) {
+    addErrorInfo(error, { type: 'buildbotClientConnection' })
+    throw error
+  }
+}
+
+const closeBuildbotClient = async function(buildbotClient) {
+  if (buildbotClient === undefined) {
+    return
+  }
+  try {
+    await promisify(buildbotClient.end.bind(buildbotClient))()
+  } catch (error) {
+    addErrorInfo(error, { type: 'buildbotClientConnection' })
+    throw error
+  }
+}
+
+const writePayload = async function(buildbotClient, payload) {
+  if (buildbotClient === undefined) {
+    return
+  }
+  try {
+    await promisify(buildbotClient.write.bind(buildbotClient))(JSON.stringify(payload))
+  } catch (e) {
+    const error = new Error(`Error writing TCP payload to buildbot: ${e.message}`)
+    addErrorInfo(error, { type: 'buildbotClient ' })
+    throw error
+  }
+}
+
+const getNextParsedResponsePromise = async function(buildbotClient) {
+  if (buildbotClient === undefined) {
+    return
+  }
+  const data = await pEvent(buildbotClient, 'data')
+  try {
+    return JSON.parse(data)
+  } catch (e) {
+    const error = new Error(`Invalid TCP payload received from the buildbot: ${e.message}\n${data}`)
+    addErrorInfo(error, { type: 'buildbotClient' })
+    throw error
+  }
+}
+
+const deploySiteWithBuildbotClient = async function(client) {
+  const payload = { action: 'deploySite' }
+  try {
+    const nextResponsePromise = getNextParsedResponsePromise(client)
+    await writePayload(client, payload)
+    const response = await nextResponsePromise
```

Comment:
We should do:

```js
const [response] = await Promise.all([
  getNextParsedResponsePromise(client),
  writePayload(client, payload),
])
```

Otherwise, if `nextResponsePromise` is rejected right away, it would wait for `writePayload()` to complete, which would lead to an [`unhandledRejection`](https://nodejs.org/api/process.html#process_event_unhandledrejection) event followed by [`rejectionHandled`](https://nodejs.org/api/process.html#process_event_rejectionhandled).